packages/core/src/data-module/subscription-store/subscriptionStore.ts

```diff
@@ -48,19 +55,36 @@ export default class SubscriptionStore<Query extends DataStreamQuery> {
       if ('queries' in subscription) {
         subscription.fulfill();
 
+        let refreshRate =
+          subscription.queries
+            .filter(isQueryWithRequestSettings)
+            .find((q) => q.requestSettings.refreshRate != null)?.requestSettings
+            .refreshRate || subscription.request.settings?.refreshRate;
```

Comment:
Nit: Consider pulling this handling into its own method to improve readability of `addSubscription`.